### DISCUSSION TITLE
Diarization Delay on livestream audio

### DISCUSSION THREAD

**Discussion by jdheeter** [deepgram/community#773] [General help]
### Which Deepgram product feature are you using?

Deepgram API - STT Streaming

### Details

Currently we have an application that can take a pre-recorded phone call audio file as input, using the Deepgram API returns the full transcription with speaker annotations. We are adding livestreaming capability, but I'm finding a weird quirk with the live API. When streaming the exact same audio on the live API, all speech is ascribed to speaker 0 for the first 20-30 seconds of the conversation. After this point the speaker annotations start working as I expect. This causes the start of each livestream session to have incorrect speaker annotations.

### If possible, please attach your code or paste it into the text box.

  dgLive = dg.listen.live({
    language: "en",
    punctuate: true,
    smart_format: true,
    model: "nova-2",
    interim_results: false,
    diarize: true,
    multichannel: false
  })
  
    const { result, error } = await deepgram.listen.prerecorded.transcribeFile(
    fs.readFileSync(fileName),
    {
      model: "nova-2-phonecall",
      diarize: true,
      detect_entities: false,
      summarize: false
    }
  )

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by jdheeter** [reply]
POST /v1/listen?language=en&punctuate=false&smart_format=true&model=nova-2&interim_results=true&diarize=true&multichannel=false
dcb8bc03-e617-43b7-96fa-3419deae0a21
1cc936a9-7c4e-4112-9c11-0fa72e4fd670
7f144d44-3e63-4f5f-a0ef-57f51d6c9fcb

**Comment by jpvajda** [comment]
@jdheeter  I'm curious if you tried this suggestion from our help center yet?

https://deepgram.gitbook.io/help-center/faq/improving-diarization-by-prepending-audio-from-the-primary-speaker

**Reply by davidvonthenen** [reply]
Another suggestion is modify the mechanism in which the audio file is generated to put every speaker on their own audio channel within the file.

**Reply by gergomiklos-seon** [reply]
Even if I get a 30 secs pre-recorded audio from the main speaker, how can I prepend audio for a live stream?
